borg/hashindex.pyx

```diff
@@ -212,7 +221,10 @@ cdef class ChunkIndex(IndexBase):
             raise KeyError(key)
         cdef int32_t refcount = _le32toh(data[0])
         if refcount != self.max_ref:
-            refcount -= 1
+            if self.max_ref - refcount < 1:
```

Comment:
note that you'll still need to compare refcount to <= 0 otherwise X - -2 = X+2 anyway and at 2**30 maxref we have plenty of time to still go positive and be bigger than 1.